ghastly/ghastly.py

```diff
@@ -440,3 +443,280 @@ def write_pour_main(pour_filename, sim_block, variable_filename, x_b, y_b, z_b,
+    recirculation simulation is the file with a name matching input_file.
+    '''
+    input_block = read_input.InputBlock(input_file)
+    sim_block = input_block.create_obj()
+
+    x_b, y_b, z_b = find_box_bounds(sim_block)
+
+    write_variable_block(var_fname, input_block, sim_block)
+
+    active_core = (sim_block.core_intake | 
+                         sim_block.core_main | 
+                         sim_block.core_outtake)
+
+    act_reg_fnames, act_reg_names = write_region_blocks(active_core)
+    
+    main_cyl = {key:value for key, value in sim_block.core_main.items() if
+                'cyl' in key}
+    out_cyl = {key:value for key, value in sim_block.core_outtake.items() if
+               'cyl' in key}
+    _, out_params = out_cyl.popitem()
+    r_chute = out_params.r
+    v_reg_fname = "v_regs.txt"
+    v_reg_name = "v_reg"
+    write_v_regs(main_cyl, r_chute, v_reg_fname, v_reg_name)
+
+    if sim_block.fidelity == 1:
+        write_recircf1_main(recirc_fname, recirc_temp, var_fname, 
+                        act_reg_fnames, act_reg_names,
+                        v_reg_fname, v_reg_name, 
+                        init_bed_fname, sim_block, x_b, y_b, z_b)
+    elif sim_block.fidelity == 2:
+        outlet_zone = {key: value for key, value in 
+                       sim_block.recirc.items() if 'out' in key}
+        outlet_fname = 'f2-recirc-zone.txt'
+        outlet_name = list(outlet_zone.keys())[0]
+        write_recircf2_regs(outlet_zone, outlet_fname)
+
+        write_recircf2_main(recirc_fname, recirc_temp, var_fname, 
+                            act_reg_fnames, act_reg_names, 
+                            v_reg_fname, v_reg_name,
+                            outlet_fname, outlet_name, 
+                            init_bed_fname, sim_block, x_b, y_b, z_b)
+
+def write_recircf2_regs(outlet_zone, outlet_fname):
+    '''
+    Writes file for the recirculation region used to select which pebbles
+    should be recirculated each loop in LAMMPS.  This region is non-physical.
+
+    Parameters
+    ----------
+    outlet_zone : dict
+        dict where the key: value pair is the 
+        name of the recirc outlet zone : Ghastly Core object that defines
+        this region.
+    outlet_fname : str
+        file name of the recirculation outlet zone region block.
+
+    Returns
+    ----------
+    This function has no return, but does generate a file named outlet_fname.
+    '''
+
+    outlet_template = env.get_template("recircf2_outlet_template.txt")
+    name, param = outlet_zone.popitem()
+    outlet_text = outlet_template.render(outlet_name=name,
+                                         x_c=param.x_c,
+                                         y_c=param.y_c,
+                                         r=param.r,
+                                         z_min=param.z_min,
+                                         z_max=param.z_max)
+    with open(outlet_fname, mode='w') as f:
+        f.write(outlet_text)
```

Comment:
It's best practice to return something, though not required. Can I suggest returning outlet_text?

```suggestion
        f.write(outlet_text)
  return outlet_text

```

https://realpython.com/python-return-statement/#returning-values-vs-modifying-globals